gateway/src/main/java/com/lyft/data/gateway/handler/HeaderBasedProxyHandler.java

```diff
@@ -39,7 +39,13 @@ protected String rewriteTarget(HttpServletRequest request) {
     String targetLabel = request.getHeader(TARGET_HEADER);
     String targetUri = "";
     if (targetLabel == null) {
-      targetUri = gatewayBackendManager.getActiveBackends().get(0).getProxyTo();
```

Comment:
It's unclear to me when this "pick first active backend" logic will be used and what should happen if the request is scheduled but there is no scheduled backend (which is allowed and request falls back to adhoc one) to pick in this function. The way it is lead to NullPointerException.